src/arrayMethodJoin.js

```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const sep = separator === undefined ? ',' : separator;
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i];
+
+      if (element === null || element === undefined) {
```

Comment:
The condition 'element === null || element === undefined' can be simplified to 'element == null'.